sdk/tests/NodeAddressTest.cc

```diff
@@ -0,0 +1,151 @@
+/*-
+ *
+ * Hedera C++ SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "impl/NodeAddress.h"
+#include "exceptions/IllegalStateException.h"
+#include "impl/IPv4Address.h"
+
+#include <gtest/gtest.h>
+#include <proto/basic_types.pb.h>
+#include <string>
+
+using namespace Hedera::internal;
+
+class NodeAddressTest : public ::testing::Test
+{
+protected:
+  [[nodiscard]] inline const int64_t& getTestNodeId() const { return mTestNodeId; }
+  [[nodiscard]] inline const int& getTestPortTLS() const { return mTestPortTLS; }
+  [[nodiscard]] inline const int& getTestPortPlain() const { return mTestPortPlain; }
+  [[nodiscard]] inline const std::string& getTestRSAPublicKey() const { return mTestRSAPublicKey; }
+  [[nodiscard]] inline const std::string& getTestIpAddress() const { return mTestIpAddress; }
+  [[nodiscard]] inline const std::string& getTestDescription() const { return mDescription; }
+  [[nodiscard]] const std::string getTestNodeAddress() const
+  {
+    const int testPortTLS = getTestPortTLS();
+    std::stringstream outputStream;
+    outputStream << getTestIpAddress() << ":" << std::to_string(testPortTLS);
+    return outputStream.str();
+  }
```

Comment:
This is known at compile-time, you could probably just declare a separate variable and return that by `const` reference.

Reply:
Done in d71ff3a1ac4f0b3de01ca053c23d011b96abcc6e.